src/app/auth/auth.effects.ts

```diff
@@ -1,38 +1,51 @@
 import { Injectable } from '@angular/core';
 import { Actions, createEffect, ofType } from '@ngrx/effects';
-import { AuthActions } from './action-types';
-import { tap } from 'rxjs';
+import { catchError, map, of, switchMap, tap } from 'rxjs';
 import { Router } from '@angular/router';
+import { AUTH_ACTIONS } from './auth.actions';
+import { HttpErrorResponse } from '@angular/common/http';
+import { AuthService } from '../guard/auth.service';
+import { LocalStorageService } from '../services/localstorage.service';
 
 @Injectable()
 export class AuthEffects {
-  login$ = createEffect(
-    () =>
-      this.actions$.pipe(
-        ofType(AuthActions.login),
-        tap(({ user }) => {
-          localStorage.setItem('user', JSON.stringify(user));
-          localStorage.setItem('accessToken', JSON.stringify(user.login_token));
-          localStorage.setItem(
-            'refreshToken',
-            JSON.stringify(user.refresh_token)
-          );
-        })
-      ),
-    { dispatch: false }
+  public login$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(AUTH_ACTIONS.login),
+      switchMap(loginData =>
+        this.authService.logIn(loginData).pipe(
+          map(response => {
+            this.router.navigateByUrl('/home');
```

Comment:
using tap for both navigation and dispatching the action throws errors so I used tap for the navigation since that is a side effect and used the map for the action dispatch
I believe I'm doing the right this